src/realm/alloc_slab.cpp

```diff
@@ -726,15 +726,22 @@ ref_type SlabAlloc::attach_file(const std::string& file_path, Config& cfg)
     // the session initiator. Another process may have the session initiator.
 
     m_file_mappings->m_file.open(path.c_str(), access, create, 0); // Throws
-    if (cfg.encryption_key)
+    size_t size_without_considering_encryption = m_file_mappings->m_file.get_size();
```

Comment:
I think `size_without_considering_encryption` is a bit difficult to quickly understand. How about simply `physical_file_size` or something